copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopyBookSource.java

```diff
@@ -0,0 +1,291 @@
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Map<String, String>> input, Emitter<StructuredRecord> emitter)
+    throws Exception {
+    Map<String, String> value = input.getValue();
+    Schema outputSchema;
+    if (config.schema == null) {
+      outputSchema = getSchema(value.keySet());
+    } else {
+      outputSchema = config.parseSchema();
+    }
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    for (Schema.Field field : outputSchema.getFields()) {
+      String fieldName = field.getName();
+      if (value.containsKey(fieldName)) {
+        builder.set(fieldName, parseValue(field, value.get(fieldName)));
+      }
+    }
+    emitter.emit(builder.build());
+  }
+
+  /**
+   * Convert each field to data type as specified by user in the output schema
+   *
+   * @param field Schema.Field object defining the field name and properties
+   * @param value value of field to be converted to datatype specified by user
+   * @return value casted to the required data type
+   * @throws IOException
+   */
+  private Object parseValue(Schema.Field field, String value) throws IOException {
+    Schema.Type fieldType = field.getSchema().getType();
+    if (fieldType.equals(Schema.Type.UNION)) {
+      fieldType = field.getSchema().getUnionSchema(0).getType();
+    }
+    switch (fieldType) {
+      case NULL:
+        return null;
+      case INT:
+        return Integer.parseInt(value);
+      case DOUBLE:
+        return Double.parseDouble(value);
+      case BOOLEAN:
+        return Boolean.parseBoolean(value);
+      case LONG:
+        return Long.parseLong(value);
+      case FLOAT:
+        return Float.parseFloat(value);
+      case BYTES:
+        return value.getBytes();
+      case STRING:
+        return value;
+    }
+    throw new IOException(String.format("Unsupported schema: %s for field: \'%s\'",
+                                        field.getSchema(), field.getName()));
+  }
+
+  /**
+   * Create schema from list field names retrieved from CopybookFiles
+   *
+   * @param filedsSet set of field names extracted from Copybook
+   * @return output schema fields
+   */
+  private Schema getSchema(Set<String> filedsSet) {
+    List<Schema.Field> fields = Lists.newArrayList();
+    for (String field : filedsSet) {
+      fields.add(Schema.Field.of(field, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    }
+    return Schema.recordOf("record", fields);
+  }
+
+  /**
+   * Config class for CopybookSource.
+   */
+  public static class CopybookSourceConfig extends ReferencePluginConfig {
+
+    @Name("binaryFilePath")
+    @Description("Complete path of the .bin to be read(Eg : hdfs://10.222.41.31:9000/test/DTAR020_FB.bin, " +
+      "file:///home/cdap/cdap/DTAR020_FB.bin).\n This will be a fixed length binary format file," +
+      " that macthes the copybook." +
+      "\n(This is done to accept files present on remote hdfs location)")
```

Comment:
```
@Description("Complete path of the .bin to be read; for example: 'hdfs://10.222.41.31:9000/test/DTAR020_FB.bin' or " +
  "'file:///home/cdap/DTAR020_FB.bin'.\n " +
  "This will be a fixed-length binary format file that matches the copybook.\n" +
  "(This is done to accept files present on a remote HDFS location.)")